pkg/metadata/image_metadata.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"encoding/json"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata/store"
+
+	runtime "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+// TODO(mikebrow): Handle versioning around all marshal/unmarshal.
+// version and versionedImageMetadata is not used now, but should be used
+// in the future:
+// 1) Only versioned metadata should be written into the store;
+// 2) Only unversioned metadata should be returned to the user;
+// 3) A conversion function is needed to convert any supported versioned
+// metadata into unversioned metadata.
+
+// imageMetadataVersion is current version of image metadata.
+const imageMetadataVersion = "v1" // nolint
+
+// versionedImageMetadata is the internal struct representing the versioned
+// image metadata
+// nolint
+type versionedImageMetadata struct {
+	// Version indicates the version of the versioned image metadata.
+	Version string `json:"version,omitempty"`
+	ImageMetadata
+}
+
+// ImageMetadata is the unversioned image metadata.
+type ImageMetadata struct {
+	// Id of the image. Normally the Digest
+	ID string `json:"id,omitempty"`
+	// Other names by which this image is known.
+	RepoTags []string `json:"repo_tags,omitempty"`
+	// Digests by which this image is known.
+	RepoDigests []string `json:"repo_digests,omitempty"`
+	// Size of the image in bytes. Must be > 0.
+	Size uint64 `json:"size,omitempty"`
+	// Authentication configuration for pulling the image.
+	Auth *runtime.AuthConfig `json:"auth,omitempty"`
+	// CreatedAt is the created timestamp.
+	CreatedAt int64 `json:"created_at,omitempty"`
+}
```

Comment:
is that different than runtime.AuthConfig.username?

I suppose we need to talk about what should be stored or not in the image metadata .. what is needed/expected vs what is nice to have.